# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStoreNotifier.h
# Commit: 0d57b435d872
# Full Hash: 0d57b435d872751c2a26879fb04bf8745314e37a
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 8 - CookieStore API - e10s support, r=smaug,cookie-reviewers,timhuang
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D217574
# ==============================================================================

diff -r a63a5bcb0524 -r 0d57b435d872 dom/cookiestore/CookieStoreNotifier.h
--- a/dom/cookiestore/CookieStoreNotifier.h	Wed Sep 11 16:26:57 2024 +0000
+++ b/dom/cookiestore/CookieStoreNotifier.h	Wed Sep 11 16:26:57 2024 +0000
@@ -9,6 +9,7 @@
 
 #include "nsIObserver.h"
 #include "mozilla/OriginAttributes.h"
+#include "mozilla/MoveOnlyFunction.h"
 
 class nsISerialEventTarget;
 
@@ -38,6 +39,8 @@
   void AddObserversOnMainThread(bool aPrivateBrowsing);
   void RemoveObserversOnMainThread(bool aPrivateBrowsing);
 
+  void DispatchEvent(const CookieListItem& aItem, bool aDeletedEvent);
+
   // Raw pointer because this object is kept alive by this CookieStore object.
   CookieStore* mCookieStore;
 